src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+          ...newNextColumnWidths,
+        },
+      },
+    };
+  }
+
+  if (action.type === actions.columnDoneResizing) {
+    return {
+      ...newState,
+      columnResizing: {
+        ...newState.columnResizing,
+        startX: undefined,
+        isResizingColumn: undefined,
+      },
+    };
+  }
+  return newState;
+};
+
+const getColumnWidths = (
+  headerIdWidths: [string, number][],
+  deltaPercentage: number,
+) => {
+  const columnWidths: Record<string, number> = {};
+  headerIdWidths.forEach(([headerId, headerWidth]) => {
+    columnWidths[headerId] = Math.max(
+      headerWidth + headerWidth * deltaPercentage,
+      0,
+    );
+  });
+  return columnWidths;
+};
+
+const isNewColumnWidthsValid = <T extends Record<string, unknown>>(
+  columnWidths: Record<string, number>,
+  headers: ColumnInstance<T>[] | undefined,
+) => {
+  // Prevents from going outside the column bounds
+  if (Object.values(columnWidths).some((width) => width <= 1)) {
+    return false;
+  }
+
+  for (const [headerId, width] of Object.entries(columnWidths)) {
+    const header = headers?.find((h) => h.id === headerId);
+    if (!header) {
+      continue;
+    }
+
+    const minWidth = header.minWidth || 0;
+    const maxWidth = header.maxWidth || Infinity;
+    if (width < minWidth || width > maxWidth) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+const useInstanceBeforeDimensions = <T extends Record<string, unknown>>(
+  instance: TableInstance<T>,
+) => {
+  const {
+    flatHeaders,
+    getHooks,
+    state: { columnResizing },
+  } = instance;
+
+  const getInstance = useGetLatest(instance);
+
+  flatHeaders.forEach((header, index) => {
+    const resizeWidth = columnResizing.columnWidths[header.id];
+    header.width = resizeWidth || header.width || header.originalWidth;
+    header.isResizing = columnResizing.isResizingColumn === header.id;
+
+    const headerToResize = header.disableResizing
+      ? getPreviousResizableHeader(header, instance)
+      : header;
+    const nextResizableHeader = getNextResizableHeader(header, instance);
+
+    header.canResize =
+      header.disableResizing != null ? !header.disableResizing : true;
+    // Show resizer when header is resizable or when next header is resizable
+    // and there is resizable columns on the left side of the resizer.
+    header.isResizerVisible =
+      header.canResize ||
+      (headerToResize && instance.flatHeaders[index + 1]?.canResize);
+
+    header.getResizerProps = makePropGetter(getHooks().getResizerProps, {
+      instance: getInstance(),
+      header: headerToResize,
+      nextHeader: nextResizableHeader,
+    });
+  });
+};
+
+const getPreviousResizableHeader = <T extends Record<string, unknown>>(
+  headerColumn: ColumnInstance<T>,
+  instance: TableInstance<T>,
+) => {
+  const headerIndex = instance.flatHeaders.findIndex(
+    (h) => h.id === headerColumn.id,
+  );
+  return [...instance.flatHeaders]
+    .slice(1, headerIndex)
```

Comment:
why start index 1?

Reply:
It was a "hack" to omit wrapper column that is required for the table by accident. Changed to firstly look at the same parent if missing then at all columns.